File: docs/dtls-implementations/nodejs/dtls_server.js
var dtls = require( 'node-dtls' );
var fs = require( 'fs' );

var key = fs.readFileSync( 'cert.key' );
var crt = fs.readFileSync( 'cert.crt' );

var server = dtls.createServer({ type: 'udp4', key: key, cert: crt });
server.bind( 5683 );

console.log('DTLS Server listening on port 5683');

server.on( 'secureConnection', function( socket ) {

  console.log( 'New connection from ' +
    [ socket.rinfo.address, socket.rinfo.port ].join(':') );
    socket.send("foda-se\n");

    socket.on( 'message', function( message ) {
    // Echo the message back
    console.log(message.toString());
  });
});

File: docs/dtls-implementations/python/dtls_server.py
import ssl
from socket import socket, AF_INET, SOCK_DGRAM
from dtls import do_patch
do_patch()
sock = ssl.wrap_socket(socket(AF_INET,SOCK_DGRAM),keyfile="cert.key",certfile="cert.crt")
sock.bind(("localhost",5683))
sock.listen(0)
print "DTLS Server listening on port 5683"
while True:
  conn,addr = sock.accept()
  data = conn.recv(1024)
  print "message: {}".format(data)

